gen/generate_test.go

```diff
@@ -79,6 +79,62 @@ func TestGenerateWithRelativePaths(t *testing.T) {
 	}
 }
 
+func TestGenerateWithHyphenPaths(t *testing.T) {
+	outputDir, err := ioutil.TempDir("", "thriftrw-generate-test")
+	require.NoError(t, err)
+	defer os.RemoveAll(outputDir)
+
+	thriftRoot, err := os.Getwd()
+	require.NoError(t, err)
+
+	tests := []struct {
+		filepath      string
+		generateError string
+		compileError  string
+	}{
+		{
+			filepath:     "internal/tests/thrift/nestedfiles_error/include_hyphen_files.thrift",
+			compileError: "including hyphenated thrift file is not allowed",
+		},
+		{
+			filepath: "internal/tests/thrift/abc-defs.thrift",
+		},
+		{
+			filepath:      "internal/tests/thrift/nestedfiles_error/abc-defs-nested.thrift",
+			generateError: "is colliding with existing file",
+		},
+		{
+			filepath: "internal/tests/thrift/nestedfiles_error/abc_defs_nested.thrift",
+		},
+	}
+
+	for _, test := range tests {
+		module, err := compile.Compile(test.filepath)
+		if test.compileError != "" {
+			assert.Error(t, err)
```

Comment:
`require.Error`

if there's no error here, the next line will panic

Reply:
It's expected to error, yes, but if we ended up changing some functionality in thriftrw that causes this test to fail, it's better to have it fail early instead of panic